pkg/costmodel/metrics.go

```diff
@@ -125,97 +134,154 @@ var (
+			toRegisterGV = append(toRegisterGV, gpuAllocGv)
+		}
+
+		if _, ok := disabledMetrics["pod_pvc_allocation"]; !ok {
+			pvAllocGv = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+				Name: "pod_pvc_allocation",
+				Help: "pod_pvc_allocation Bytes used by a PVC attached to a pod",
+			}, []string{"namespace", "pod", "persistentvolumeclaim", "persistentvolume"})
+			toRegisterGV = append(toRegisterGV, pvAllocGv)
+		}
+
+		if _, ok := disabledMetrics["kubecost_network_zone_egress_cost"]; !ok {
+			networkZoneEgressCostG = prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "kubecost_network_zone_egress_cost",
+				Help: "kubecost_network_zone_egress_cost Total cost per GB egress across zones",
+			})
+			toRegisterGage = append(toRegisterGage, networkZoneEgressCostG)
+		}
+
+		if _, ok := disabledMetrics["kubecost_network_region_egress_cost"]; !ok {
+			networkRegionEgressCostG = prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "kubecost_network_region_egress_cost",
+				Help: "kubecost_network_region_egress_cost Total cost per GB egress across regions",
+			})
+			toRegisterGage = append(toRegisterGage, networkRegionEgressCostG)
+		}
+
+		if _, ok := disabledMetrics["kubecost_network_internet_egress_cost"]; !ok {
+			networkInternetEgressCostG = prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "kubecost_network_internet_egress_cost",
+				Help: "kubecost_network_internet_egress_cost Total cost per GB of internet egress.",
+			})
+			toRegisterGage = append(toRegisterGage, networkInternetEgressCostG)
+		}
+
+		if _, ok := disabledMetrics["kubecost_cluster_management_cost"]; !ok {
+			clusterManagementCostGv = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+				Name: "kubecost_cluster_management_cost",
+				Help: "kubecost_cluster_management_cost Hourly cost paid as a cluster management fee.",
+			}, []string{"provisioner_name"})
+			toRegisterGV = append(toRegisterGV, clusterManagementCostGv)
+		}
+
+		if _, ok := disabledMetrics["kubecost_load_balancer_cost"]; !ok {
+			lbCostGv = prometheus.NewGaugeVec(prometheus.GaugeOpts{ // no differentiation between ELB and ALB right now
+				Name: "kubecost_load_balancer_cost",
+				Help: "kubecost_load_balancer_cost Hourly cost of load balancer",
+			}, []string{"ingress_ip", "namespace", "service_name"}) // assumes one ingress IP per load balancer
+			toRegisterGV = append(toRegisterGV, lbCostGv)
+		}
 
 		// Register cost-model metrics for emission
-		prometheus.MustRegister(cpuGv, ramGv, gpuGv, gpuCountGv, totalGv, pvGv, spotGv)
-		prometheus.MustRegister(ramAllocGv, cpuAllocGv, gpuAllocGv, pvAllocGv)
-		prometheus.MustRegister(networkZoneEgressCostG, networkRegionEgressCostG, networkInternetEgressCostG)
-		prometheus.MustRegister(clusterManagementCostGv, lbCostGv)
+		for i := range toRegisterGV {
+			prometheus.MustRegister(toRegisterGV[i])
+		}
+		for i := range toRegisterGage {
+			prometheus.MustRegister(toRegisterGage[i])
+		}
```

Comment:
Since you have these slices of emitters, what if you checked the disabled condition in these loops, rather than hardcoding an if block for each one?

So, something like
```go
for _, gv := range toRegisterGV {
    if _, ok := disabledMetrics[gv.Name]; ok {
        continue
    }

    prometheus.MustRegister(gv)
}
```

Reply:
Instead of doing a range loop, I also think you could just do something like:

```go
prometheus.MustRegister(toRegisterGV...)
```

Reply:
I tried both of these actually, but I didn't get either to work so I just opted to do it the dumb way.

Specifically, `MustRegister` apparently won't take `[]*Prometheus.GaugeVec` as an argument even though it'll take several `*Prometheus.GaugeVec`. Also, I don't think you can get the name out of the Gaugevec that easily.

Reply:
Annoying that you can't get the name out of the Gaugevec. Bolt's suggestion should work, though. It's a standard Go method for passing a slice of things to a function that takes variadic args. See https://gobyexample.com/variadic-functions

Reply:
I understand that, I'm saying that when I tried using `toRegisterGV...` it gives me an IncompatibleAssign.

Reply:
Darn